### PR TITLE
Move breadcrumb json configuration to viewmodel

### PR DESCRIPTION
**Pull request by diedburn**
### Description
Move breadcrumb json configuration to viewmodel and serialize it using Magento json serializer
Tweak code motivated by discussion in pull request #15162 

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=15521) <br/>All committers have signed the CLA.

**Comment by diedburn** [comment]
I signed the license agreement.

**Comment by magento-engcom-team** [comment]
@diedburn thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php [hunk 1]
```diff
@@ -27,18 +28,26 @@ class Breadcrumbs extends DataObject implements ArgumentInterface
...
         Data $catalogData,
-        ScopeConfigInterface $scopeConfig
+        ScopeConfigInterface $scopeConfig,
+        Json $json
     ) {
         parent::__construct();
 
         $this->catalogData = $catalogData;
         $this->scopeConfig = $scopeConfig;
+        $this->json = $json;
```
Comment (VladimirZaets) [inline app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php]: Hi, @diedburn, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't add the required dependency to the constructor method.
thanks

**Comment by VladimirZaets** [comment]
Hi @vgelani, please, sign CLA, otherwise, we can't process your pull request

**Comment by gelanivishal** [comment]
@VladimirZaets I have signed.

**Comment by dmanners** [comment]
Hi @diedburn it still says that you have not signed the CLA, could you please check this for us so we can continue with this PR. Often this can happen if the github user and committed user have a different email address.

**Comment by diedburn** [comment]
@dmanners I tried signing the CLA again, the email it was committed with was added to my github user the day I created the pull request.

**Comment by diedburn** [comment]
@dmanners 
I got it working by changing my primary email, then signing the CLA.

**Comment by VladimirZaets** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, here is your [Magento instance](https://pr-15521.engcom.dev.magento.com).
Admin access: [https://pr-15521.engcom.dev.magento.com/admin](https://pr-15521.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Please make sure you are PR **author** or **assignee** to access the instance.

**Comment by magento-engcom-team** [comment]
Hi @diedburn. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.6 release.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**